storage/innobase/row/row0import.cc

```diff
@@ -4333,6 +4335,37 @@ fil_tablespace_iterate(
 	return(err);
 }
 
+static void row_import_autoinc(dict_table_t *table, row_prebuilt_t *prebuilt,
+                               uint64_t autoinc)
+{
+  if (!table->persistent_autoinc)
+  {
+    ut_ad(!autoinc);
+    return;
+  }
+
+  if (autoinc)
+  {
+    btr_write_autoinc(dict_table_get_first_index(table), autoinc - 1);
+  autoinc_set:
+    table->autoinc= autoinc;
+    sql_print_information("InnoDB: %`.*s.%`s autoinc value set to " UINT64PF,
+                          int(table->name.dblen()), table->name.m_name,
+                          table->name.basename(), autoinc);
+  }
+  else if (TABLE *t= prebuilt->m_mysql_table)
+  {
+    if (const Field *ai= t->found_next_number_field)
+    {
+      autoinc= 1 +
+        btr_read_autoinc_with_fallback(table, innodb_col_no(ai),
```

Comment:
With a copy of `tt.ibd` and without this fix, I got the following:
```
mysqltest: At line 91: query 'INSERT INTO t0t VALUES(NULL)' failed: 167: Out of range value for column 'id' at row 1
```
For debugging I had temporarily added the following, which shows the out-of-range `TINYINT` value 129:
```
SHOW CREATE TABLE t0t;
Table	Create Table
t0t	CREATE TABLE `t0t` (
  `id` tinyint(4) NOT NULL AUTO_INCREMENT,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB AUTO_INCREMENT=129 DEFAULT CHARSET=latin1 COLLATE=latin1_swedish_ci
```
In the final test I will use `INSERT` and `SELECT` to avoid any result differences in future versions that might change the output of `SHOW CREATE TABLE`.